components/datadog/agentparams/filepermissions/permissions_win.go

```diff
@@ -0,0 +1,64 @@
+package filepermissions
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+	"github.com/DataDog/test-infra-definitions/common"
+)
+
+type WindowsPermissionsOption = func(*WindowsPermissions) error
+
+// WindowsPermissions contains the information to configure the permissions of a file on Windows.
+type WindowsPermissions struct {
+	// RemoveDefaultPermissions removes the default permissions from the file. This is useful when you want to set permissions for secrets management.
+	RemoveDefaultPermissions bool
+
+	// If you are familiar with the icacls command, you can provide a custom command directly.
+	IcaclsCommand string
+}
+
+var _ FilePermissions = (*WindowsPermissions)(nil)
+
+// NewWindowsPermissions creates a new WindowsPermissions object and applies the given options.
+func NewWindowsPermissions(options ...WindowsPermissionsOption) optional.Option[FilePermissions] {
+	p, err := common.ApplyOption(&WindowsPermissions{}, options)
+
+	if err != nil {
+		panic("Could not create WindowsPermissions: " + err.Error())
+	}
+	return optional.NewOption[FilePermissions](p)
+}
+
+// SetupPermissionsCommand returns a command that sets the permissions of a file. It relies on the icacls command.
+func (p *WindowsPermissions) SetupPermissionsCommand(path string) string {
+	cmd := ""
+	if p.RemoveDefaultPermissions {
+		cmd = fmt.Sprintf(`icacls "%v" /remove (Get-LocalUser) "Everyone" "Users" "Administrators" "SYSTEM" "Authenticated Users" "CREATOR OWNER" /inheritance:r /t /c /l;`, path)
```

Comment:
Feel free to give suggestions on how to do it better 😅

Reply:
if I understand correctly, `icacls.exe /reset` will remove all explicit ACE and leave the file with only inherited ACE. Then you can use an option to decide if you want to enable or disable inheritance.

i think `icacls.exe /reset` followed by `icacls.exe /inheritance:r ...` will leave the file with only the explicit permissions specified by the `/grant` or `/deny` parameters.

if you use `get-acl path | fl` you can examine the SDDL and the ACE with the `ID` option are the inherited ACE.

Reply:
You're right but the goal here is to remove everything including permissions specified by the `/grant` or `/deny` that's why I'm using `remove` instead of `reset`. I'm using `/reset` in ResetPermissionsCommand for the use case you described

Reply:
Here, `/reset` should work in most cases (especially in our secrets e2e tests) but I  wasn't sure if we could `grant`/`deny` new permissions by default in which case `/reset` might have failed 🤷

Reply:
`/reset` will remove all explicit permissions "`/grant` or `/deny" )

Reply:
You're right, I just checked and `/inheritance:r` does the job. I'll change it, thanks !

Reply:
I'll replace this option by `DisableInheritance` since we don't have use cases where this would not be enough yet.
We will be able to add `/reset` later if needed. Does it sound good to you ?

Reply:
the only files with explicit permissions tests are likely to encounter are ones that they themselves create. so as long as each test uses a new file or `/reset` the permissions some other way (`ResetPermissionsCommand` ? ) inbetween tests it should be fine.

Reply:
I've made some modifications 7acdbecdcd90610d411b07eebf47d7f036e41417